nav2_waypoint_follower/src/waypoint_follower.cpp

```diff
@@ -127,6 +127,11 @@ WaypointFollower::followWaypoints()
     get_logger(), "Received follow waypoint request with %i waypoints.",
     static_cast<int>(goal->poses.size()));
 
+  if (goal->poses.size() == 0) {
+    action_server_->terminate_all();
```

Comment:
I think this should auto-succeed rather than terminate since it technically succeeded in doing nothing